### ISSUE TITLE
Why <=> returns false for invalid comparison?

### ISSUE THREAD

**Issue by franklinyu** [gshutler/useragent#39]
If `<=>` returns nil for invalid comparison, we can reuse the standard [`Comparable`](https://ruby-doc.org/core-2.3.1/Comparable.html) module, which has been available since at least 1.9.3. Any use case for that? I think most people would just use other derived operators, instead of directly use `<=>`.

**Comment by gshutler** [comment]
It looks like the custom `Comparable` module comes from the first commit so I'm not sure why it's done that way.

Changing it for a `nil` returning implementation breaks several specification so it's not a simple fix but I think it is a good idea to explore.

**Comment by franklinyu** [comment]
Yes, it is not expected to be a simple fix. I think it may even break backward compatibility (if anyone is really using this operator). Hard choice.
